tests/v2/validation/snapshot/snapshot_restore_k8s_upgrade_test.go

```diff
@@ -64,6 +68,24 @@ func (s *SnapshotRestoreK8sUpgradeTestSuite) TestSnapshotRestoreK8sUpgrade() {
 	}
 
 	for _, tt := range tests {
+		clusterID, err := clusters.GetV1ProvisioningClusterByName(s.client, s.client.RancherConfig.ClusterName)
+		require.NoError(s.T(), err)
+
+		cluster, err := tt.client.Steve.SteveType(clusters.ProvisioningSteveResourceType).ByID(clusterID)
+		require.NoError(s.T(), err)
+
+		updatedCluster := new(apisV1.Cluster)
```

Comment:
just a note; funny we found different ways to get the k8s version lol. I used apiv1.clusterSpec -> cluster.Spec.